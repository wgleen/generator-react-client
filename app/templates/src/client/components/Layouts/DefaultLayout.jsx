import React from 'react'
import PropTypes from 'prop-types'

const DefaultLayout = (props) => {
  const { children } = props

  return (
    <div>
      {children}
    </div>
  )
}

DefaultLayout.defaultProps = {
  children: null
}

DefaultLayout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string
  ])
}

export default DefaultLayout
