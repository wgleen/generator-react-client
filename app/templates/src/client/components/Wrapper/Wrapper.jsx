import React from 'react'
import { Provider } from 'react-redux'
import {
  store
} from '../../lib/store'

const Wrapper = props => (
  <Provider store={store}>
    {props.children}
  </Provider>
)

export default Wrapper