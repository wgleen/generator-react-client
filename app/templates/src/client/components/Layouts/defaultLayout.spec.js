import React from 'react'
import { mount } from 'enzyme'
import Wrapper from '../Wrapper/Wrapper'
import DefaultLayout from './DefaultLayout'

describe('<DefaultLayout />', () => {
  it('should render correctly', () => {
    let component = mount(
      <Wrapper>
        <DefaultLayout />
      </Wrapper>
    )

    expect(component).toMatchSnapshot()
    
    component.unmount()
  })
})