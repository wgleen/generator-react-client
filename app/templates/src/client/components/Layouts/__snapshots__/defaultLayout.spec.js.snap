// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLayout /> should render correctly 1`] = `
<Wrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <DefaultLayout>
      <div />
    </DefaultLayout>
  </Provider>
</Wrapper>
`;
